catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -13,6 +13,1727 @@
+        String destination = TEST_DESTINATION;
+        Metacard metacard = getTestMetacard();
+
+        metacard.setAttribute(new AttributeImpl(Metacard.TAGS,
+                Collections.singletonList(RegistryConstants.REGISTRY_TAG)));
+
+        Set<String> destinations = new HashSet<>();
+        destinations.add(destination);
+
+        federationAdminServiceImpl.addRegistryEntry(metacard, destinations);
+
+        verify(catalogFramework, never()).create(any(CreateRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testAddRegistryEntryWithInvalidMetacardNoRegistryTag() throws Exception {
+        String registryId = RegistryObjectMetacardType.REGISTRY_ID;
+        String destination = TEST_DESTINATION;
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                registryId));
+
+        Set<String> destinations = new HashSet<>();
+        destinations.add(destination);
+
+        federationAdminServiceImpl.addRegistryEntry(metacard, destinations);
+
+        verify(catalogFramework, never()).create(any(CreateRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testAddRegistryEntryWithIngestException() throws Exception {
+        String registryId = RegistryObjectMetacardType.REGISTRY_ID;
+        String destination = TEST_DESTINATION;
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                registryId));
+
+        Set<String> destinations = new HashSet<>();
+        destinations.add(destination);
+
+        when(catalogFramework.create(any(CreateRequest.class))).thenThrow(IngestException.class);
+
+        federationAdminServiceImpl.addRegistryEntry(metacard, destinations);
+
+        verify(catalogFramework, never()).create(any(CreateRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testAddRegistryEntryWithSourceUnavailableException() throws Exception {
+        String registryId = RegistryObjectMetacardType.REGISTRY_ID;
+        String destination = TEST_DESTINATION;
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                registryId));
+
+        Set<String> destinations = new HashSet<>();
+        destinations.add(destination);
+
+        when(catalogFramework.create(any(CreateRequest.class))).thenThrow(SourceUnavailableException.class);
+
+        federationAdminServiceImpl.addRegistryEntry(metacard, destinations);
+
+        verify(catalogFramework, never()).create(any(CreateRequest.class));
+    }
+
+    @Test
+    public void testAddRegistryEntryMetacard() throws Exception {
+        String registryId = RegistryObjectMetacardType.REGISTRY_ID;
```

Comment:
Can you consider doing something like `getRegistryMetacard()` to cut down on some of the duplication between these tests. Seems like they have a lot of the same code. Particularly

`Metacard metacard = getTestMetacard();
 metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
               registryId));
  metacard.setAttribute(new AttributeImpl(Metacard.TAGS,
                Collections.singletonList(RegistryConstants.REGISTRY_TAG)));`

You may find greater efficiency and maintainability looking at what is common across some of these tests.